### PR TITLE
feat: add data loader framework

### PR DESCRIPTION
**Pull request by bzp2010**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

This PR is the basis for the refactored data import functionality, which provides the key `Import`/`Export` interface definition and `DataSets` data interface so that we can implement different data import implementations, such as OpenAPI or Swagger.

This is part of the refactoring, and a refactoring of the OpenAPI codec will be submitted later.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Review by nic-chen — COMMENTED** [review]
Please update the description of the PR, I don't quite understand the background of the PR and what the end goal is to achieve, thanks.

File: api/internal/handler/data_loader/loader/loader.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package loader
+
+import "github.com/apisix/manager-api/internal/core/entity"
+
+type DataSets struct {
```
Comment (starsz) [inline api/internal/handler/data_loader/loader/loader.go:21]: Better to add desc for the struct.

**Comment by bzp2010** [comment]
> Please update the description of the PR, I don't quite understand the background of the PR and what the end goal is to achieve, thanks.

Updated

----

This PR is the basis for the refactored data import functionality, which provides the key Import/Export interface definition and DataSets data interface so that we can implement different data import implementations, such as OpenAPI or Swagger.

File: api/internal/handler/data_loader/loader/loader.go (hunk 1 above)
Comment (bzp2010) [reply api/internal/handler/data_loader/loader/loader.go:25]: @starsz Updated

**Review by nic-chen — APPROVED** [review]

**Comment by bzp2010** [comment]
ping @starsz

**Review by starsz — APPROVED** [review]

**Comment by starsz** [comment]
> ping @starsz

LGTM.

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**